### PR TITLE
Parser: Support SHOW PROFILE Syntax

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 

Support `SHOW PROFILE` Syntax for TiDB Parser. See: https://dev.mysql.com/doc/refman/5.7/en/show-profile.html for details.

### What is changed and how it works?
1. Add related syntax in `parser.y`
2. Add a new show type `ShowProfile` for `ast.ShowStmt`
3. Add related fields for `SHOW PROFILE` in `ShowStmt`
4. Update `ShowStmt.Restore` to support `SHOW PROFILE` syntax.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: ast/dml.go [hunk 1]
```diff
@@ -1899,6 +1917,47 @@ func (n *ShowStmt) Restore(ctx *RestoreCtx) error {
...
+		ctx.WriteKeyWord("PROFILE")
+		if len(n.ShowProfileTypes) > 0 {
+			for i, tp := range n.ShowProfileTypes {
+				ctx.WritePlain(" ")
+				if i != 0 {
+					ctx.WritePlain(",")
+				}
+				switch tp {
+				case ProfileTypeCPU:
+					ctx.WritePlain("CPU")
```
Comment (kennytm) [inline ast/dml.go]:
```suggestion
					ctx.WriteKeyWord("CPU")
```

change the rest too.

File: parser_test.go [hunk 2]
```diff
@@ -681,6 +681,10 @@ func (s *testParserSuite) TestDBAStmt(c *C) {
 		{`SHOW EVENTS FROM test_db WHERE definer = 'current_user'`, true, "SHOW EVENTS IN `test_db` WHERE `definer`='current_user'"},
 		{`SHOW PLUGINS`, true, "SHOW PLUGINS"},
 		{`SHOW PROFILES`, true, "SHOW PROFILES"},
+		{`SHOW PROFILE`, true, "SHOW PROFILE"},
+		{`SHOW PROFILE FOR QUERY 1`, true, "SHOW PROFILE FOR QUERY 1"},
+		{`SHOW PROFILE CPU FOR QUERY 2`, true, "SHOW PROFILE CPU FOR QUERY 2"},
+		{`SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1`, true, "SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1"},
```
Comment (kennytm) [inline parser_test.go]: Please add a test case involving multiple profiles

```sql
show profile cpu, memory, block io, context switches, page faults, ipc, swaps, source limit 100
```

File: ast/dml.go (hunk 1 above)
Comment (spongedu) [reply ast/dml.go]: yes, should be `WriteKeyWord` here. I'll fix them

File: parser_test.go (hunk 2 above)
Comment (spongedu) [reply parser_test.go]: ok

**Comment by spongedu** [comment]
@kennytm  PTAL :)

**Comment by spongedu** [comment]
Build break after merge `master`

**Comment by spongedu** [comment]
@tiancaiamao PTAL :)

**Comment by spongedu** [comment]
@zz-jason @lysu PTAL

**Comment by spongedu** [comment]
@morgo  PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
Please resolve conflicts @spongedu

**Comment by spongedu** [comment]
@tiancaiamao conflicts resolved. PTAL

**Pull request merged**